### ISSUE TITLE
Modal transition assertions

### ISSUE THREAD

**Issue by defagos** [SRGSSR/srgidentity-apple#9]
@oiledCode reported that modal transition asserts when the account view is presented from a sheet modal view (default iOS 13 presentation style).

The code does not crash in release builds, but the [work recently done on modals](https://github.com/SRGSSR/playsrg-apple/issues/123) was not applied to SRG Identity. This should be fixed.

### Issue type

Incorrect behavior

### Environment information

* Application version: 3.0.1
* iOS: Any
* Device: Any

### Reproducibility

Always reproducible

**Comment by defagos** [comment]
I fixed the issue on the `feature/modal-presentation-fix` branch:

- The modal presentation style is now custom, as should be.
- I updated the transition code accordingly.
- To ensure the presenting view is correctly animated (i.e. that we have a `fromView`), I introduced a `UIPresentationController` subclass to enabled presenting view removal. I updated the transition code to reinstall the view when the modal is dismissed.
